CMakeLists.txt

```diff
@@ -93,13 +93,13 @@ elseif( CMAKE_CXX_COMPILER MATCHES ".*/hipcc$" )
   set( CMAKE_C_COMPILE_OPTIONS_VISIBILITY_INLINES_HIDDEN "-Xcompiler ${CMAKE_C_COMPILE_OPTIONS_VISIBILITY_INLINES_HIDDEN}" )
   set( CMAKE_CXX_COMPILE_OPTIONS_VISIBILITY_INLINES_HIDDEN "-Xcompiler ${CMAKE_CXX_COMPILE_OPTIONS_VISIBILITY_INLINES_HIDDEN}" )
 elseif( CMAKE_CXX_COMPILER MATCHES ".*/hcc$" )
-  message( STATUS "HCC compiler set; ROCm backend selected [ CXX=/opt/rocm/bin/hcc cmake ... ]" )
+  message( STATUS "HCC compiler set; ROCm backend selected [ CXX=/opt/rocm-<ver>/bin/hcc cmake ... ]" )
 endif( )
 
 # This finds the rocm-cmake project, and installs it if not found
 # rocm-cmake contains common cmake code for rocm projects to help setup and install
 set( PROJECT_EXTERN_DIR ${CMAKE_CURRENT_BINARY_DIR}/extern )
-find_package( ROCM CONFIG QUIET PATHS /opt/rocm )
+find_package( ROCM CONFIG QUIET PATHS ${ROCM_PATH} /opt/rocm )
```

Comment:
The clear and better way to use ${ROCM_PATH} is:
check ROCM_PATH has been set by user or not, if not set it to /opt/rocm only once in the root CMakelist.
As well, message the information of the ROCM_PATH set by user or not and the exact path.
And remove all the below "/opt/rocm" from any search path.

Reply:
Moreover, ROCM_PATH is unnecessary; we can just use CMAKE_PREFIX_PATH.

Reply:
@feizheng10 ,
Default value of ${ROCM_PATH} is not set as it is not required, if not defined/set the search path will include only /opt/rocm. 

@malcolmroberts,
Reason for not adding into CMAKE_PREFIX_PATH was to give highest priority to the user provided ROCm path as part of the find_package parameter, so that we are sure which ROCm version is used to refer to hcc/hip, etc. Going further multiple ROCm can be installed on a machine and when compiling any libs, we need to sure how the dependencies are met.

Reply:
The user can just put the path to the ROCm cmake as the first element in the CMAKE_PREFIX_PATH; this guarantees that the user-specified path has highest priority.  

There is no reason to add another environment variable; things already work without this patch.  Moreover, not using standard cmake variables will almost certainly cause problems down the road.

Do you have a test environment with multiple rocm installs where you can try the branch that I provided?

Reply:
Before these changes, CMAKE_PREFIX_PATH was not used for searching all dependencies, the cmake file had hard coded /opt/rocm path at all place. So one of thought was to keep it in similar way so added ROCM_PATH (/opt/rocm-ver) to the search patch list. 
Will make the changes based on CMAKE_PREFIX_PATH and verify the running and get back.